libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/ConfigureRouterTask.kt

```diff
@@ -2,23 +2,20 @@ package com.mapbox.services.android.navigation.v5.navigation
 
 import android.os.AsyncTask
 import com.mapbox.navigator.Navigator
-import com.mapbox.navigator.TileEndpointConfiguration
+import com.mapbox.navigator.RouterParams
+import com.mapbox.services.android.navigation.v5.internal.navigation.HttpClient
 
 internal class ConfigureRouterTask(
     private val navigator: Navigator,
-    private val tilePath: String,
-    private val tileEndpointConfiguration: TileEndpointConfiguration,
+    private val routerParams: RouterParams,
     private val callback: OnOfflineTilesConfiguredCallback
 ) : AsyncTask<Void, Void, Long>() {
 
     @Synchronized
     override fun doInBackground(vararg paramsUnused: Void): Long =
         navigator.configureRouter(
-            tilePath,
-            null,
-            null,
-            2, // Two threads for downloading tiles simultaneously
-            tileEndpointConfiguration
+            routerParams,
+            HttpClient(routerParams.endpointConfig!!.userAgent, true)
```

Comment:
If `TileEndpointConfiguration` is `null` this is going to 💥 Should we check here and send empty `userAgent` as we're 👀 that downstream (`HttpClient#init`). In any case, we know for sure it's going to be `@NonNull` as we're creating an instance upstream (`OfflineNavigator#configure`).

Reply:
routerParams.endpointConfig can't be null here, we build it on a previous step.
Also we can't pass empty `userAgent`. We have a check that throws exception if userAgent is empty

Reply:
Yeah I know, I'm trying to avoid the situation in which `OfflineNavigator` code changes and sends a `null` and then we get an unexpected NPE instead of the expected `"Empty UserAgent is not allowed" `IllegalStateException`. What do you think?

Reply:
ah, ok, pass empty string and get readable error is better )

Reply:
fixed